internal/pkg/aws/ec2/ec2_test.go

```diff
@@ -158,6 +158,93 @@ func TestEC2_ListVPC(t *testing.T) {
 	}
 }
 
+func TestEC2_ListAZs(t *testing.T) {
+	testCases := map[string]struct {
+		mockClient func(m *mocks.Mockapi)
+
+		wantedErr string
+		wantedAZs []AZ
+	}{
+		"return wrapped error on unexpected call error": {
+			mockClient: func(m *mocks.Mockapi) {
+				m.EXPECT().DescribeAvailabilityZones(gomock.Any()).Return(nil, errors.New("some error"))
+			},
+			wantedErr: "describe availability zones: some error",
+		},
+		"returns AZs that are available and opted-in": {
+			mockClient: func(m *mocks.Mockapi) {
+				m.EXPECT().DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{
+					Filters: []*ec2.Filter{
+						{
+							Name:   aws.String("zone-type"),
+							Values: aws.StringSlice([]string{"availability-zone"}),
+						},
+						{
+							Name:   aws.String("state"),
+							Values: aws.StringSlice([]string{"available"}),
+						},
+					},
+				}).Return(&ec2.DescribeAvailabilityZonesOutput{
+					AvailabilityZones: []*ec2.AvailabilityZone{
```

Comment:
Should we add one or two that get filtered out?

Reply:
Luckily the filtering doesn't happen client side ✌️ ! EC2 should return to us only AZs that are available so we don't have to do any filtering ourselves

Reply:
Ah, gotcha gotcha. Right, makes sense. 👍🏼